src/rdiff_backup/SetConnections.py

```diff
@@ -180,19 +183,26 @@ def _init_connection(remote_cmd):
     conn_number = len(Globals.connections)
     conn = connection.PipeConnection(stdout, stdin, conn_number)
 
-    _check_connection_version(conn, remote_cmd)
+    if not _validate_connection_version(conn, remote_cmd):
+        return None
```

Comment:
Why returning `None` ? Shouldn't we stop the process with a fatal error ? At least I would raise an exception instead and let the caller handle the exception.

Reply:
First, I've always learned that exceptions take more time than normal if/then/else clauses, so it plays against performance and should be used sparingly. Second, see my comment previously, this is how I can test more than one server at once. And, third, I also wanted to cleanly quit the code and not somewhere in the middle.